internal/pkg/otel/run.go

```diff
@@ -80,3 +86,23 @@ func newSettings(version string, configPaths []string) (*otelcol.CollectorSettin
 		DisableGracefulShutdown: true,
 	}, nil
 }
+
+func ensureRegistryExists() error {
+	storageDir := os.Getenv("STORAGE_DIR")
+	if storageDir == "" {
+
+		// by default use "${path.data}/registry/otelcol" to store offsets
+		storageDir = filepath.Join(paths.Data(), "registry", "otelcol")
```

Comment:
I think we probably need some discussion on what our directory structure should be, considering https://github.com/elastic/ingest-dev/issues/3671 and https://github.com/elastic/elastic-agent/issues/5304 for example.

There are also things besides filelog that may want disk persistent, who don't use a registry concept. If we needed more than this, where would they go in the directory structure? Here? What if we decided we needed more than one storage extension?